server/src/main/java/com/unosquare/carmigo/security/AppUser.java

```diff
@@ -0,0 +1,51 @@
+package com.unosquare.carmigo.security;
+
+import static com.unosquare.carmigo.constant.AppConstants.NOT_PERMITTED;
+
+import com.unosquare.carmigo.exception.UnauthorizedException;
+import java.util.ArrayList;
+import lombok.Builder;
+import lombok.Getter;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.context.annotation.RequestScope;
+
+@Configuration
+public class AppUser {
+
+  @Bean
+  @RequestScope
+  public Current get() {
+    final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+    if (authentication.getPrincipal() instanceof CustomUserDetails) {
+      final CustomUserDetails customUserDetails = (CustomUserDetails) authentication.getPrincipal();
+      return Current.builder()
+          .id(customUserDetails.getId())
+          .username(customUserDetails.getUsername())
+          .userAccessStatus(getUserAccessStatus(customUserDetails))
+          .build();
+    }
+    throw new UnauthorizedException(NOT_PERMITTED);
```

Comment:
I'm not sure I would throw an exception in a Bean definition.  Does this work before you are authenticated?

Reply:
I followed the snippet you sent me. This `throw` is because of the `if` statement and the "if" is to avoid a `NullPointerException`